gpflow/kernels/categorical.py

```diff
@@ -0,0 +1,97 @@
+from typing import Any, Optional
+
+import numpy as np
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from .. import default_int, set_trainable
+from ..base import Parameter, TensorType
+from . import Kernel
+
+tfd = tfp.distributions
+st = tfp.bijectors.Softplus().inverse
+
+
+def latent_from_labels(Z: tf.Tensor, labels: TensorType) -> TensorType:
+    """
+    Converts a tensor of labels into a tensor of the latent space values of those labels.
+    :param Z: Z is a tensor whose entries are the latent space values of the relevant label
```

Comment:
The param descriptions don't seem to correspond to the arguments!  I'm guessing `labels` is supposed to be `[..., N]` and the output `[..., N, label_dim]`, which suggests `Z` is `[num_labels, label_dim]`? Or maybe I'm getting confused?

Also maybe call it `X_labels` or something rather than `labels` so it's obvious that it's part of the data?